lib/rules/require-quoted-keys-in-objects.js

```diff
@@ -44,7 +44,7 @@ module.exports.prototype = {
     check: function(file, errors) {
         file.iterateNodesByType('ObjectExpression', function(node) {
             node.properties.forEach(function(prop) {
-                if (prop.shorthand || prop.method || prop.kind !== 'init') {
+                if (prop.shorthand || prop.method || prop.kind !== 'init' || prop._babelType === 'SpreadProperty') {
```

Comment:
Why not an AST class that contains tree-iterator logic and getBabelType?
On Aug 10, 2015 9:12 AM, "Oleg Gaidarenko" <notifications@github.com> wrote:

> In lib/rules/require-quoted-keys-in-objects.js
> <https://github.com/jscs-dev/node-jscs/pull/1649#discussion_r36628665>:
>
> > @@ -44,7 +44,7 @@ module.exports.prototype = {
> >      check: function(file, errors) {
> >          file.iterateNodesByType('ObjectExpression', function(node) {
> >              node.properties.forEach(function(prop) {
> > -                if (prop.shorthand || prop.method || prop.kind !== 'init') {
> > +                if (prop.shorthand || prop.method || prop.kind !== 'init' || prop._babelType === 'SpreadProperty') {
>
> utils.getBabelType might best since, we take just that, not another
> node.type or anything else
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jscs-dev/node-jscs/pull/1649/files#r36628665>.
>